src/utils/fetchers.ts

```diff
@@ -24,179 +24,147 @@
 
  */
 
-import { useState, useEffect, useContext } from "react"
+import { useContext } from "react"
+import { useQuery, useMutation, useQueryClient } from 'react-query'
```

Comment:
This file is the heart of the change; Most of the changes in this PR are resultant from consolidation of logic in this file.